modules/standard/List.chpl

```diff
@@ -796,6 +816,25 @@ module List {
       return result;
     }
 
+    proc const ref last() ref {
```

Comment:
If you're up to it, I'd like to try refactoring all the overloads of `first` and `last` to use a helper method (in order to avoid code duplication).

Reply:
The signatures might look something like this:

```chapel
proc const ref _firstHelper() ref { ...

proc const ref _lastHelper() ref { ...
```

Reply:
@dlongnecke-cray How do we convert from const ref to ref? Does the helper internally create a copy from const ref input and return reference to that copy?